lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -2193,6 +2229,40 @@ ParseResult FIRStmtParser::parseIntegerLiteralExp(Value &result) {
   return success();
 }
 
+/// list-exp ::= list-type '(' exp* ')'
+ParseResult FIRStmtParser::parseListExp(Value &result) {
+  auto loc = getToken().getLoc();
+  FIRRTLType type;
+  if (parseListType(type))
+    return failure();
+  auto listType = type_cast<ListType>(type);
+  auto elementType = listType.getElementType();
+
+  if (parseToken(FIRToken::l_paren, "expected '(' in List expression"))
+    return failure();
+
+  SmallVector<Value, 3> operands;
+  if (parseListUntil(FIRToken::r_paren, [&]() -> ParseResult {
+        Value operand;
+        locationProcessor.setLoc(loc);
+        if (parseExp(operand, "expected expression in List expression"))
```

Comment:
Following FIRRTL spec, we maybe want to have a particular notion of "Property Expr" (like RefExp) instead of taking any expression -- to only parse for it where supported/appropriate (presently, RHS of a propassign, plus as elements of List "literal"), and to enforce the "static reference" requirement for expressions given to propassign.

Something to clean up, especially as these land in the spec.